OneSignalSDK/unittest/src/test/java/com/test/onesignal/NotificationOpenedActivityHMSIntegrationTestsRunner.java

```diff
@@ -68,27 +70,50 @@ public void beforeEachTest() throws Exception {
         ShadowOSUtils.supportsHMS(true);
     }
 
-    private static Intent helper_baseHMSOpenIntent() {
+    private static @NonNull Intent helper_baseHMSOpenIntent() {
         return new Intent()
                 .setFlags(Intent.FLAG_ACTIVITY_NO_HISTORY | Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_MULTIPLE_TASK)
                 .setAction("android.intent.action.VIEW");
     }
 
-    private static void helper_startHMSOpenActivity(@NonNull Intent intent) {
-        Robolectric.buildActivity(NotificationOpenedActivityHMS.class, intent).create();
+    private static @NonNull Intent helper_basicOSHMSOpenIntent() throws JSONException {
+        return helper_baseHMSOpenIntent()
+                .putExtra(
+                    "custom",
+                    new JSONObject() {{
+                        put("i", UUID.randomUUID().toString());
+                }}.toString()
+        );
     }
 
-    private static void helper_initSDKAndFireHMSNotificationOpenIntent() throws Exception {
-        OneSignal.init(RuntimeEnvironment.application, "123456789", ONESIGNAL_APP_ID);
-        fastColdRestartApp();
-
-        Intent intent = helper_baseHMSOpenIntent()
+    private static @NonNull Intent helper_basicOSHMSOpenIntentWithActionId(final @NonNull String actionId) throws JSONException {
+        return helper_baseHMSOpenIntent()
                 .putExtra(
                         "custom",
                         new JSONObject() {{
                             put("i", UUID.randomUUID().toString());
+                            put(BUNDLE_KEY_ACTION_ID, actionId);
                         }}.toString()
                 );

```

Comment:
Use the `OSNotificationFormatHelper.PAYLOAD_OS_NOTIFICATION_ID` for `"i"`?
Use the `OSNotificationFormatHelper.PAYLOAD_OS_ROOT_CUSTOM` for `"custom"`?